File: Chapter-5/Insertion_sort.cpp
#include<iostream>

using namespace std;

int main(){
    int n;
    cout<<" enter the size of an array: ";
    cin>>n;
    int arr[n];
    for(int i=0;i<n;i++){
        cout<<" enter elements : ";
        cin>>arr[i];

    }

    for(int j=0;j<n;j++){
        for(int i=0;i<n-1;i++){
            if(arr[i]>arr[i+1]){
                int temp = arr[i];
                arr[i] = arr[i+1];
                arr[i+1] = temp;
            }
        }
    }

    cout<<"The Insertion Sort is: "<<" ";
    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    return 0;
}

File: Chapter-5/Insertion_sort.py
def Insertion_Sort(arr):
    for i in range(1,len(arr)):
        temp = arr[i]

        while arr[i-1] > temp and i>0:
            arr[i],arr[i-1] = arr[i-1],arr[i]
            i = i-1

    print(arr)

arr = []
num = int(input("enter the number of elements in array : "))
print("enter the elements: ")
for i in range(0,num):
    ele = int(input())
    arr.append(ele)

print("The Insertion sort array is: ",end=" ")
Insertion_Sort(arr)